frontend/src/scenes/project/Settings/ActorOnEventsQuerying.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { useActions, useValues } from 'kea'
+import { teamLogic } from 'scenes/teamLogic'
+import { LemonSwitch, LemonTag } from '@posthog/lemon-ui'
+
+export function ActorOnEventsQuerying(): JSX.Element {
+    const { updateCurrentTeam } = useActions(teamLogic)
+    const { currentTeam } = useValues(teamLogic)
+
+    return (
+        <div id="querying">
+            <h2 className="subtitle">
+                Optimized Querying{' '}
+                <LemonTag type="warning" style={{ marginLeft: 8 }}>
+                    Beta
+                </LemonTag>
+            </h2>
+            <p>
+                This enables optimized querying. More details regarding how these queries work can be found{' '}
```

Comment:
I feel we should be stronger with the messaging here. Something along the lines of:

Optimized querying is coming, but this will destroy some existing insights.

We'll be rolling this out to everyone by default soon, so here's your chance to test & let us know of problems beforehand?

cc: @joethreepwood  for better ways of saying this^.

The idea:
1. We want to give people control to test this out.
2. So that when we release it to everyone, it doesn't come as a brutal surprise that some insights have stopped working / there's data mismatch. Will definitely happen to all users depending on `alias` to retroactively link distinctIDs.

Reply:
yeah 100%. I left as placeholder and forgot to comment

Reply:
I have [an announcement pending](https://github.com/PostHog/posthog.com/pull/4101) which we could hypothetically link to as well. It'd be better to have this in the docs, but the current page we're linking to isn't obviously relevant to this toggle @pjhul 

I'd avoid language like 'will destroy some existing insights' -- it's a bit alarming and isn't specific about the effect. 

Q: Will affected insights work again if the toggle is flipped back?

```suggestion
                Optimize queries using person properties. Some existing insights may not work while this is enabled due 
                to {' '} <a href="https://posthog.com/docs/how-posthog-works/queries">how queries work</a>.
```
Maybe?

Reply:
> Q: Will affected insights work again if the toggle is flipped back?

For now, yes; but not when we start the rollout on cloud.

Reply:
Alternate based on what Joe's saying: Let's not go into any details (link for that), but only mention how users are going to be affected?

"We're changing how insights are calculated. This will rollout to everyone soon, but you can test it on your own by turning on this setting" 

.. Hmm, no wait.

Reply:
@EDsCODE @rcmarron thinking about it a bit more, I don't think the opt-in banner & project setting makes sense anymore.

Here's why: 

1. We're going to be rolling it out forcefully to everyone in X days.
2. We're not sure exactly how many people will face issues here. (last time when we rolled out to 15%, one person complained, and we rolled back, worrying if it was a code issue. Turns out, it was an expected discrepancy.)
3. We want to hear about issues from users as soon as they occur.

By making it an action they can take, 50% (say), might not take it. All of these people will be caught unaware by the rollout, and also delay us, since they'll reach out to us about their issues at the final rollout stage, vs now.

So, I propose:

1. We have a banner on insights saying something like: "Results looking different? We've changed how insights are calculated. Read more about this here. Reach out to us if you see something unexpected".
2. This is shown only to those teams who are within the rollout.
3. We rollout using our feature flag.

This ensures we're never in the dark, users who face issues immediately reach out to us, AND aren't left wondering why things _seem_ broken. (and also massively speeds up adoption on cloud).